app/src/main/java/com/example/movieapp/AppModule.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.movieapp
+
+import com.example.movieapp.data.MovieLocalDataSource
+import com.example.movieapp.data.MovieRemoteDataSource
+import com.example.movieapp.data.MovieRepository
+import com.example.movieapp.ui.MainViewModel
+import com.example.movieapp.ui.SplashViewModel
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+
+val appModule = module {
+    single {
+        MovieLocalDataSource()
+    }
+    viewModel {
+        SplashViewModel(get())
+    }
+    viewModel {
+        MainViewModel(get())
+    }
+    single<String> (named("BASE_URL")) {
+        "https://api.themoviedb.org/"
```

Comment:
There isn't really a need for a hard coded url string as a dependency.
Try providing it in a singleton (object) `Constants.kt` instead, or even better grandle build config(enviroment) variable.

Reply:
Done